dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,626 @@
+/*
+    DSSCollection contains collection group & completion functionality for DSS.
+    Author: Jeremy Ahrens jer.ahrens@dapperlabs.com
+*/
+
+import NonFungibleToken from "./NonFungibleToken.cdc"
+import MetadataViews from 0xMETADATAVIEWSADDRESS
+
+// The DSSCollection contract
+//
+pub contract DSSCollection: NonFungibleToken {
+
+    // Contract Events
+    //
+    pub event ContractInitialized()
+
+    // NFT Collection Events
+    //
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+
+    // Events
+    //
+    pub event CollectionGroupCreated(
+        id: UInt64,
+        name: String,
+        typeName: String,
+        startTime: UFix64?,
+        endTime: UFix64?,
+        timeBound: Bool
+    )
+    pub event CollectionGroupClosed(id: UInt64)
+    pub event SlotCreated(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        logicalOperator: String,
+        typeName: String,
+        slotType: String
```

Comment:
Sorry, remind me what this one does.  TypeName tells us what contract it belongs to, and itemID would be what part of that contract (eg edtion.id, edition.play.id, edition.play.metadata.playerName etc), but what's slotType again?

Reply:
typeName is the contract it is a part of. I have it here to enable multiple contracts being a part of the same collection group. Aka ticket master NFT + NFL All day. It will be A.address.NFT.

slotType will be edition.id, play.id, etc.

Reply:
yeah, but then you can't and if it's stuck with one, see my comment below.

Reply:
Up do you, could make it more expensive to compute what you have if they are different, but caching could still help so w/e.  I don't have a strong opinion here.  It seems more flexible, but potentially harder to do right.

Reply:
I have done the refactor. Slot contains and array of items. typeName is now on the item level. Am leaving typeName at the slot level.